application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,417 @@
+     */
+    private Optional<ZonedDateTime> getLatestPostDateFromItems(List<Item> items,
+            String dateFormatterPattern) {
+        return items.stream()
+            .map(item -> getDateTimeFromItem(item, dateFormatterPattern))
+            .max(ZonedDateTime::compareTo);
+    }
+
+    /**
+     * Posts an RSS item to a text channel.
+     *
+     * @param textChannels the text channels to which the item will be posted
+     * @param rssItem the RSS item to post
+     * @param feedConfig the RSS feed configuration
+     */
+    private void postItem(List<TextChannel> textChannels, Item rssItem, RSSFeed feedConfig) {
+        MessageEmbed embed = constructEmbedMessage(rssItem, feedConfig).build();
+        textChannels.forEach(channel -> channel.sendMessageEmbeds(List.of(embed)).queue());
+    }
+
+    /**
+     * Updates the last posted date to the database for the specified RSS feed configuration.
+     * <p>
+     * This will insert a <b>new</b> entry to the database if the provided {@link RssFeedRecord} is
+     * null.
+     *
+     * @param feedConfig the RSS feed configuration
+     * @param rssFeedRecord the record representing the RSS feed, can be null if not found in the
+     *        database
+     * @param lastPostedDate the last posted date to be updated
+     *
+     * @throws DateTimeParseException if the date cannot be parsed
+     */
+    private void updateLastDateToDatabase(RSSFeed feedConfig, @Nullable RssFeedRecord rssFeedRecord,
+            ZonedDateTime lastPostedDate) {
+        DateTimeFormatter dateTimeFormatter =
+                DateTimeFormatter.ofPattern(feedConfig.dateFormatterPattern());
+        String lastDateStr = lastPostedDate.format(dateTimeFormatter);
+
+        if (rssFeedRecord == null) {
+            database.write(context -> context.newRecord(RSS_FEED)
+                .setUrl(feedConfig.url())
+                .setLastDate(lastDateStr)
+                .insert());
+            return;
+        }
+
+        database.write(context -> context.update(RSS_FEED)
+            .set(RSS_FEED.LAST_DATE, lastDateStr)
+            .where(RSS_FEED.URL.eq(feedConfig.url()))
+            .executeAsync());
+    }
+
+    /**
+     * Attempts to get a {@link ZonedDateTime} from an {@link Item} with a provided string date time
+     * format.
+     * <p>
+     * If either of the function inputs are null or a {@link DateTimeParseException} is caught, the
+     * oldest-possible {@link ZonedDateTime} will get returned instead.
+     *
+     * @param item The {@link Item} from which to extract the date.
+     * @param dateTimeFormat The format of the date time string.
+     * @return The computed {@link ZonedDateTime}
+     * @throws DateTimeParseException if the date cannot be parsed
+     */
+    private static ZonedDateTime getDateTimeFromItem(Item item, String dateTimeFormat)
+            throws DateTimeParseException {
+        String pubDate = item.getPubDate().orElse(null);
```

Comment:
I would say this is a bad approach. It doesn't really make sense using Optionals if you just use `.orElse(null)` and then check if its null. Instead, store the Optional and check if its empty.

Reply:
Done!